src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,301 @@
+package duke;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Date;
+
+import javafx.stage.Stage;
+
+/**
+ * Parser class is used for making sense of all possible user commands.
+ *
+ * @author Tseng Chen-Yu
+ * @version %I%, %G%
+ * @since 1.0
+ */
+public class Parser {
+    private final Stage stage;
+
+    public enum CmdType {
+        bye,
+        list,
+        mark,
+        unmark,
+        delete,
+        todo,
+        deadline,
+        event,
+        find
+    }
+
+    public Parser(Stage stage) {
+        this.stage = stage;
+    }
+
+    private boolean isByeCommand(String userInput) {
+        return userInput.replaceAll("\\s", "")
+                .equals(CmdType.bye.name());
+    }
+
+    private boolean isListCommand(String userInput) {
+        return userInput.replaceAll("\\s", "")
+                .equals(CmdType.list.name());
+    }
+
+    private boolean isMarkCommand(String userInput) {
+        return (userInput.split(" ")[0])
+                .equals(CmdType.mark.name());
+    }
+
+    private boolean isUnmarkCommand(String userInput) {
+        return (userInput.split(" ")[0])
+                .equals(CmdType.unmark.name());
+    }
+
+    private boolean isDeleteCommand(String userInput) {
+        return userInput.split(" ")[0]
+                .equals(CmdType.delete.name());
+    }
+
+    private boolean isTodoCommand(String userInput) {
+        return userInput.split(" ")[0]
+                .equals(CmdType.todo.name());
+    }
+
+    private boolean isDeadlineCommand(String userInput) {
+        return userInput.split(" ")[0].equals(CmdType.deadline.name());
+    }
+
+    private boolean isEventCommand(String userInput) {
+        return userInput.split(" ")[0]
+                .equals(CmdType.event.name());
+    }
+
+    public boolean isFindCommand(String userInput) {
+        return userInput.split(" ")[0].equals(CmdType.find.name());
+    }
+
+    public String parseCommandWithResponse(String userInput) {
```

Comment:
This method seems way too long. As a suggestion, perhaps you could break your method up into handleByeCommand, handleMarkCommand etc so it's easier to debug.